### PR TITLE
Simplify flash middleware API and implementation

### PR DESCRIPTION
**Pull request by jerben**

### CODE REVIEW

File: sihl/src/web_flash.ml [hunk 1]
```diff
@@ -103,28 +100,36 @@ let decode_flash cookie_key req =
...
-      resp
-    else (* Remove flash cookie *)
+    if delete_if_not_set
+    then
+      (* Flash was not touched, don't set cookie *)
       Opium.Response.remove_cookie cookie_key resp
-  | Some flash ->
-    if Flash.equals old_flash flash
-    then (* Flash was not touched, don't set cookie *)
+    else (* Remove flash cookie *)
```
Comment (aronerben) [inline sihl/src/web_flash.ml:116]: This comment should be for the `then` branch and vice versa, correct?

**Review by aronerben — CHANGES_REQUESTED** [review]
Please check

File: sihl/src/web_flash.ml (hunk 1 above)
Comment (jerben) [reply sihl/src/web_flash.ml]: Fixed

**Review by aronerben — APPROVED** [review]
Looks good!

**Pull request merged**